### PR TITLE
Autoscale

### PR DESCRIPTION
**Pull request by amitnayak1505**

### CODE REVIEW

File: asg.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+resource "aws_launch_configuration" "ec2" {
+  image_id        = var.ami
```
Comment (vladimirmukhin) [inline asg.tf]: Please use aws ami data source instead of variable

File: asg.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  instance_type   = var.type
+  security_groups = [aws_security_group.public.id]
+  user_data       = file("userdata1.sh")
```
Comment (vladimirmukhin) [inline asg.tf:21]: the `userdata1.sh` is missing in git

File: asg.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    create_before_destroy = true
+  }
+}
+
+resource "aws_autoscaling_group" "asg" {
+  launch_configuration = aws_launch_configuration.ec2.id
+  health_check_type    = "EC2"
+  min_size             = var.asg_count
+  max_size             = var.asg_count
+  desired_capacity     = var.asg_count
```
Comment (vladimirmukhin) [inline asg.tf]: No need to parametrize this setting

File: asg.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  vpc_zone_identifier  = [aws_subnet.public[count.index].id]
```
Comment (vladimirmukhin) [inline asg.tf]: This will not work

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: asg.tf (hunk 2 above)
Comment (amitnayak1505) [reply asg.tf:21]: added to the git

File: asg.tf (hunk 3 above)
Comment (amitnayak1505) [reply asg.tf]: Made the changes

File: asg.tf (hunk 4 above)
Comment (amitnayak1505) [reply asg.tf]: updated

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**